src/woo_publications/publications/tests/test_documents_admin.py

```diff
@@ -100,23 +98,55 @@ def test_document_admin_search(self):
             self.assertContains(search_response, "field-identifier", 1)
             self.assertContains(search_response, "title one", 1)
 
+    def test_document_admin_list_filters(self):
+        self.app.set_user(user=self.user)
+
+        with freeze_time("2024-09-24T12:00:00-00:00"):
+            DocumentFactory.create(
+                officiele_titel="title one",
+                verkorte_titel="one",
+                omschrijving="Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+                creatiedatum="2024-09-24",
+            )
+        with freeze_time("2024-09-25T12:30:00-00:00"):
+            document2 = DocumentFactory.create(
+                officiele_titel="title two",
+                verkorte_titel="two",
+                omschrijving="Vestibulum eros nulla, tincidunt sed est non, facilisis mollis urna.",
+                creatiedatum="2024-09-25",
+            )
+        reverse_url = reverse("admin:publications_document_changelist")
+
         with freeze_time("2024-09-25T00:14:00-00:00"):
-            with self.subTest("filter_on_registratiedatum"):
-                today = "2024-09-25T00:00:00-00:00"
-                tomorrow = "2024-09-26T00:00:00-00:00"
-
-                search_response = self.app.get(
-                    reverse_url,
-                    {
-                        "registratiedatum__gte": today,
-                        "registratiedatum__lt": tomorrow,
-                    },
-                    user=self.user,
-                )
-
-                self.assertEqual(search_response.status_code, 200)
-                self.assertContains(search_response, "field-identifier", 1)
-                self.assertContains(search_response, publication2.identifier, 1)
+            response = self.app.get(reverse_url)
+
+        self.assertEqual(response.status_code, 200)
+
+        with self.subTest("filter_on_registratiedatum"):
+            search_response = response.click(description=_("Today"), index=0)
+
+            self.assertEqual(search_response.status_code, 200)
+
+            # Sanity check that we indeed filtered on registratiedatum
+            self.assertIn(
+                "registratiedatum", search_response.request.environ["QUERY_STRING"]
+            )
```

Comment:
`request.environ` is super low level - in django land the `HttpRequest` is the lowest we're supposed to go.

We can grab the actual django request object and make assertions on that using the familiar interfaces:

```
assert "registratiedatum__gte" in search_response.context["request"].GET
```
here we use the django template context to extract the `request` instance

additionally, I prefer using an `assert` here to make it clear that this isn't the main target of our test, but a sanity check rather than an assertion that would fail the test